internal/certprovider_test.go

```diff
@@ -2,33 +2,124 @@ package internal_test
 
 import (
 	"crypto/x509"
+	"io"
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 
 	"github.com/blocky/nitrite"
 	"github.com/blocky/nitrite/internal"
+	"github.com/blocky/nitrite/mocks"
 )
 
+func TestUnzipAWSRootCerts(t *testing.T) {
+	t.Run("happy path", func(t *testing.T) {
+		// given
+		zipped := internal.AWSNitroEnclavesRootZip
+
+		// when
+		unzipped, err := internal.UnzipAWSRootCerts(zipped)
+
+		// then
+		require.NoError(t, err)
+		assert.NotEmpty(t, unzipped)
+	})
+
+	errorTests := []struct {
+		name   string
+		zipped []byte
+	}{
+		{"empty", []byte{}},
+		{"nil", nil},
+		{"invalid", []byte("invalid zip bytes")},
```

Comment:
Given the assumptions for the zip structure and naming convention, consider adding test cases that check number of files in the zip as well as the the `root.pem` name. Would require you to add two zip files to `testdata/`. Might also make `root.pem` a constant so we can reference in source func and test

Reply:
done - I was able to create the zip file dynamically in the tests